core/src/main/scala/cats/instances/seq.scala

```diff
@@ -17,6 +20,17 @@ trait SeqInstances extends cats.kernel.instances.SeqInstances {
 
       def combineK[A](x: Seq[A], y: Seq[A]): Seq[A] = x ++ y
 
+      override def combineAllOptionK[A](as: IterableOnce[Seq[A]]): Option[Seq[A]] = {
+        val iter = as.iterator
+        if (iter.isEmpty) None else Some(appendAll(iter, Seq.newBuilder[A]).result())
```

Comment:
actually, in this pattern, won't `if (as.isEmpty) None else Some(appendAll(as.iterator, ...)` work also that way we don't need to convert to iterator in the empty case?

Reply:
Is it possible to check `as.isEmpty` on an `IterableOnce` without consuming it?

In fact, in 2.13 `IterableOnce#isEmpty` is deprecated: "(Since version 2.13.0) Use `.iterator.isEmpty` instead".

I think what we could do here is use `IterableOnce#knownSize`.

Reply:
okay. Bummer we have to allocate in order to check emptiness.

Reply:
Yeah that was what I wanted to write originally but ran into the deprecation